### ISSUE TITLE
test_prints_on_failure is flakey on CircleCI

### ISSUE THREAD

**Issue by alexwlchan** [HypothesisWorks/hypothesis-python#523]
Third time in as many days that I’ve seen the following flakey test in CircleCI:

```
____________________________ test_prints_on_failure ____________________________
[gw1] darwin -- Python 2.7.10 /Users/distiller/.cache/hypothesis-build-runtimes/.tox/pypy-full/bin/python
tests/cover/test_arbitrary_data.py:48: in test_prints_on_failure
    assert 'Draw 1: [0, 0]' in result
E   AssertionError: assert 'Draw 1: [0, 0]' in ''
----------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "/Users/distiller/hypothesis-python/tests/common/utils.py", line 51, in raises
    assert False, "Expected to raise an exception but didn't"
AssertionError: Expected to raise an exception but didn't
============== 1 failed, 1172 passed, 2 skipped in 70.39 seconds ===============
```

Only Python 2.7.10, and it’s always this test. Usually clears up after a re-run, but we should understand why it keeps failing.

**Comment by DRMacIver** [comment]
My suspicion is that what's happening is that after #511 it's sometimes triggering a *different* assertion error that's internal to Hypothesis. I'll investigate.

**Comment by Zac-HD** [comment]
Closed by #524
